### ISSUE TITLE
Compatibility notation warning can be triggered by binding a pattern

### ISSUE THREAD

**Issue by tchajed** [coq/coq#9166]
#### Version

8.10+alpha

#### Operating system

macOS

#### Description of the problem

The `[compatibility-notation,deprecated]` warning can be triggered by binding the notation's name in a pattern. The deprecated name is properly used as a pattern variable and not the notation, but the compatibility/warning system still thinks the notation is used.

As a workaround, one can simply change the bound name to anything not deprecated, but this comes from a real example in `hs-to-coq`'s [Enum.v](https://github.com/antalsz/hs-to-coq/blob/82a64710760156de0fa3720b302ae4d6a0eec41e/examples/base-src/manual/GHC/Enum.v#L319).

```
Require Import NArith.

Definition foo (x: nat) : nat :=
  match x with
  | 0 => 0
  | S Ngt => Ngt
  end.
```

**Comment by Zimmi48** [comment]
This problem already exists in 8.8 and 8.9. Fixing this for 8.9 would be nice, although in 8.10, these compatibility aliases will be mostly removed so it should resolve most instances (in practice).

**Comment by tchajed** [comment]
Sounds good. This is an extremely low priority issue for me (it's not in my own project and the workaround is simple), so waiting for compatibility notations to be removed is a great solution.

**Comment by herbelin** [comment]
I made a quick fix at #9167.
